src/compiler/interface.jl

```diff
@@ -185,7 +188,7 @@ function copy!(x::AbstractVector,  gs::Grads)
   x
 end
 
-broadcasted(f, gss::Grads...) = map(f, gss...)
+broadcasted(f, gs::Grads, gss::ADictOrGrads...) = map(f, gs, gss...)
```

Comment:
Would want this
```suggestion
broadcasted(f, gs::ADictOrGrads, gss::ADictOrGrads...) = map(f, gs, gss...)
```

But that would overload broadcasting for the case where `gs` and `gss` are just `Dict`s (no `Grads`). Don't know a quick way around it but unfortunate that the first arg must be a `Grad`.

Reply:
yep, I don't know either, unless we start adding a bunch of definitions like
```julia
broadcasted(f, gs1::AbstractDict, gs2::Grads, gss::ADictOrGrads...) = map(f, gs1, gs2, gss...)
```
and the corresponding ones for `map`, but this seems pretty annoying and I would leave it to future PRs

Reply:
Is `broadcasted(f, gss::Vararg{Union{AbstractDict, Grads}})` different from `broadcasted(f, gss::Vararg{<:Union{AbstractDict, Grads}})`? I think the first forces the union type. Let me double check.